### PR TITLE
Simplify margin logic and fix wrong alignment

### LINKED ISSUE #594: Incorrect alignment in sideline
Image 1

The issue is in the screenshot above. The sideline is aligned incorrectly while scroll horizontally to the right. (out of window)

### PR DESCRIPTION
**Pull request by jcs090218**
Calculation of `line-number-display-width` should be calculated inside `lsp-ui-sideline--margin-width` and we should not apply margin to the calculation while displaying sidelines.

This also fixes #594.

### CODE REVIEW

File: lsp-ui-sideline.el [hunk 1]
```diff
@@ -426,26 +429,26 @@ Push sideline overlays on `lsp-ui-sideline--ovs'."
...
                       (lsp-ui-sideline--split-long-lines it)))
-             (display-lines (butlast lines (- (length lines) lsp-ui-sideline-diagnostic-max-lines)))
+             (display-lines (nreverse (butlast lines (- (length lines) lsp-ui-sideline-diagnostic-max-lines))))
              (offset 1))
-        (dolist (line (nreverse display-lines))
-          (let* ((message (string-trim (replace-regexp-in-string "[\t ]+" " " line)))
-                 (len (length message))
+        (dolist (line display-lines)
+          (let* ((msg (string-trim (replace-regexp-in-string "[\t ]+" " " line)))
+                 (msg (replace-regexp-in-string " " " " msg))
```
Comment (jcs090218) [inline lsp-ui-sideline.el:436]: This is the same as 

https://github.com/emacs-lsp/lsp-ui/blob/c5b094d330c7dec3ebcdbb1e129b53f7c7b89587/lsp-ui-doc.el#L916

File: lsp-ui-sideline.el [hunk 2]
```diff
@@ -337,13 +344,9 @@ is set to t."
...
 (defun lsp-ui-sideline--window-width ()
+  "Calculate the window width only inside the text area."
   (- (min (window-text-width) (window-body-width))
-     (lsp-ui-sideline--margin-width)
-     (or (and (>= emacs-major-version 27)
-              ;; We still need this number when calculating available space
-              ;; even with emacs >= 27
-              (lsp-ui-util-line-number-display-width))
-         0)))
+     (lsp-ui-sideline--margin-width)))
```
Comment (jcs090218) [inline lsp-ui-sideline.el:349]: Move the entire line number calculation to function `lsp-ui-sideline--margin-width`.

File: lsp-ui-sideline.el [hunk 3]
```diff
@@ -296,11 +303,10 @@ CURRENT is non-nil when the point is on the symbol."
          (str (if lsp-ui-sideline-show-symbol
                   (concat info " " (propertize (concat " " symbol " ") 'face face))
                 info))
-         (len (length str))
-         (margin (lsp-ui-sideline--margin-width)))
+         (len (length str)))
     (add-face-text-property 0 len 'lsp-ui-sideline-global nil str)
     (concat
-     (propertize " " 'display `(space :align-to (- right-fringe ,(lsp-ui-sideline--align len margin))))
+     (propertize " " 'display `(space :align-to (- right-fringe ,(lsp-ui-sideline--align len))))
```
Comment (jcs090218) [inline lsp-ui-sideline.el:309]: Now we have the calculation in function `lsp-ui-sideline--margin-width`. Remove the extra margin.

File: lsp-ui-sideline.el [hunk 4]
```diff
@@ -426,26 +429,26 @@ Push sideline overlays on `lsp-ui-sideline--ovs'."
...
+                 (end-pos (lsp-ui-util-column (car pos-ov)))
+                 (string (concat (propertize (spaces-string (lsp-ui-sideline--align len end-pos)))
+                                 (propertize msg 'display (lsp-ui-sideline--compute-height))))
                  (ov (and pos-ov (make-overlay (car pos-ov) (car pos-ov)))))
             (when pos-ov
               (setq offset (1+ (car (cdr pos-ov))))
-              (overlay-put ov 'after-string string)
               (overlay-put ov 'kind 'diagnostics)
-              (overlay-put ov 'before-string " ")
+              (overlay-put ov 'before-string string)
```
Comment (jcs090218) [inline lsp-ui-sideline.el:451]: Is there a reason for this? 😕

File: lsp-ui-sideline.el [hunk 5]
```diff
@@ -563,8 +565,7 @@ from the language server."
            symbols)
       (setq lsp-ui-sideline--tag tag
             lsp-ui-sideline--last-width (window-text-width))
-      (when (and line-changed lsp-ui-sideline-show-diagnostics)
-        (lsp-ui-sideline--diagnostics buffer bol eol))
+      (lsp-ui-sideline--diagnostics buffer bol eol)
```
Comment (jcs090218) [inline lsp-ui-sideline.el:568]: Remove the check `line-changed` or else if the user scroll horizontally this doesn't take effect.

Function `lsp-ui-sideline--diagnostics` already check for the flag `lsp-ui-sideline-show-diagnostics` hence we simply remove the entire check here.

**Comment by XO39** [comment]
This totally broke the sideline and made it unusable for me, I had to downgrade to #592. I hope it gets fixed soon.

**Comment by jcs090218** [comment]
@XO39 Can you tell me what's broken? And what Emacs version you are using? Thanks!

**Comment by Yevgnen** [comment]
Same here. Almost unusable when typing because the line is moving up and down.

- lsp-ui version: `master`
- emacs version: `GNU Emacs 27.1.91 (build 2, x86_64-apple-darwin20.2.0, Carbon Version 164 AppKit 2022.2) of 2021-03-13`


Screen Shot 2021-03-25 at 11 36 38 PM

**Pull request merged**